packages/plugins/model-define/cycle-gan-model-define/src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { ModelDefineType, UniModel, ModelDefineArgsType, ImageSample, CocoDataset, ImageDataset } from '@pipcook/pipcook-core';
+import * as path from 'path';
+import * as fs from 'fs';
+
+const boa = require('@pipcook/boa');
+const { tuple } = boa.builtins();
+const sys = boa.import('sys');
+sys.path.insert(0, path.join(__dirname, '..'));
+const tf = boa.import('tensorflow');
+const { CycleGAN } = boa.import('CycleGAN.models');
+const np = boa.import('numpy');
+const cv2 = boa.import('cv2');
+const base64 = boa.import('base64');
+
+type PredictType = 'a2b' | 'b2a';
+
+let opt = {
+  // data
+  DATA_ROOT: '',         // path to images (should have subfolders 'train', 'val', etc)
+  shapeA: tuple([ 128, 128, 3 ]),
+  shapeB: tuple([ 128, 128, 3 ]),
+  resize: tuple([ 143, 143 ]),         // scale images to this size
+  crop  : tuple([ 128, 128 ]),     //  then crop to this size
+
+  // net definition
+  which_model_netD: 'basic',        // selects model to use for netD
+  which_model_netG: 'resnet_6blocks',   // selects model to use for netG
+  use_lsgan: 1,                 // if 1, use least square GAN, if 0, use vanilla GAN
+  ngf: 64,               // //  of gen filters in first conv layer
+  ndf: 64,               // //  of discrim filters in first conv layer
+  lmbd: 10.0,
+  idloss: 1.0,
+
+  // optimizers
+  lr: 0.0002,            // initial learning rate for adam
+  beta1: 0.5,            // momentum term of adam
+
+  // training parameters 
```

Comment:
it's better to allow users to specify values for some hyper-parameters such as batch-size, learning rate, etc.

Reply:
All the params can be replaced by pipeline config: `params`.